app/controllers/application_controller.rb

```diff
@@ -18,26 +18,33 @@ class ApplicationController < ActionController::API
 
   rescue_from JWTSessions::Errors::Unauthorized, with: :not_authorized
 
+  rescue_from Pundit::NotAuthorizedError, with: :forbidden
+
   private
 
-  # The current_user can be found in the JWT payload
+  # The current_user can be found in the JWT payload; eagerly load the user's roles so they can be discriminated
+  # against
   def current_user
-    @current_user ||= User.find(payload['user_id'])
+    @current_user ||= User.includes(:roles).find_by(id: payload['user_id'])
   end
 
-  # JSONAPI friendly unauthorized (401) handler
-  def not_authorized(exception)
+  # JSONAPI friendly not found (404) handler
+  def forbidden(exception)
```

Comment:
Here's where we handle any of Pundit's rejections.  Create a pleasant JSONAPI errors collection.